app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -365,12 +377,12 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
      */
     private fun setData() {
 
-        val totalUsers: Int = skillData.skillRating?.stars?.totalStar!!
-        val oneStarUsers: Int = skillData.skillRating?.stars?.oneStar!!
-        val twoStarUsers: Int = skillData.skillRating?.stars?.twoStar!!
-        val threeStarUsers: Int = skillData.skillRating?.stars?.threeStar!!
-        val fourStarUsers: Int = skillData.skillRating?.stars?.fourStar!!
-        val fiveStarUsers: Int = skillData.skillRating?.stars?.fiveStar!!
+        val totalUsers: Int = skillData.skillRating?.stars?.totalStar as Int
```

Comment:
Yes they are. But, I have already declared them as non-nullable in the response file. Then, I removed double exclamation. Android Studio showed a warning 'unsafe calls on a nullable receiver'. And, it did not allow me to perform the comparison operation. When I used explicit type casting, all warnings were removed and the app worked fine. I am not sure if I am missing something. Checking it once again.

Reply:
@iamareebjamal On removing ```: Int``` the warnings in this line vanished but then, wherever I used these values, it says "Required : Int , Found : Int?"
Also, if I use "toInt()", then the same message "unsafe calls on a nullable receiver" is shown. 
Here is the code for the response file for Stars object :

```
package org.fossasia.susi.ai.rest.responses.susi

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 *
 * Created by arundhati24 on 05/06/18.
 */
class Stars : Serializable {

    @SerializedName("one_star")
    @Expose
    var oneStar: Int = 0

    @SerializedName("two_star")
    @Expose
    var twoStar: Int = 0

    @SerializedName("three_star")
    @Expose
    var threeStar: Int = 0

    @SerializedName("four_star")
    @Expose
    var fourStar: Int = 0

    @SerializedName("five_star")
    @Expose
    var fiveStar: Int = 0

    @SerializedName("total_star")
    @Expose
    var totalStar: Int = 0

    @SerializedName("avg_star")
    @Expose
    var averageStar: Float = 0f
}
```

Reply:
Just curious, why not use val and data class. I can see that the types here are non-nullable but the reason I guess they are inferred as nullable are because of the parent object can be null. I still won't recommend the double exclamation. You may use `as Int` but there's a bigger problem here. So, use this method but create an issue to refactor things

Reply:
@iamareebjamal I agree. I revisited the official Kotlin documentation and read about Null safety, using '!!' would throw NPE, and as you had mentioned, it would be a better choice to add null checks. But, here, when I checked totalStar explicitly for null, it said "the expression is too complex". I too feel that they are being inferred as nullable because the parent object can be null. I am opening an issue for this and will try using val and data class, as you have just mentioned.